### PR TITLE
update prompt of review/revise to meet gpt3.5

### PR DESCRIPTION
**Pull request by better629**
**Features**

- add more detail to make it output as json format
- rm `indent=4` to use a compact string in prompt.
    
**Feature Docs**

**Influence**

**Result**

```
(metagpt_v2) MacBook-Pro:MetaGPT xxx$ pytest -s tests/metagpt/actions/test_action_node.py

========================================================================= 11 passed in 67.54s (0:01:07) ==========================================================================

```

**Other**

[test_indent.txt](https://github.com/geekan/MetaGPT/files/13935649/test_indent.txt)
change `test_indent.txt` to `test_indent.py`

with `Follow format example's json format` inside `action` means add extra like below
```
## action
Follow format example's json format, generate output and make sure it follows the format example.
```

test result
```

## gpt-3.5-turbo-1106

### without `Follow format example's json format` inside `action`

with_indent: True, succ_cnt: 0/10

with_indent: False, succ_cnt: 5/10

### with `Follow format example's json format` inside `action`

with_indent: True, succ_cnt: 0/10

with_indent: False, succ_cnt: 10/10

## gpt-4-1106-preview

### without `Follow format example's json format` inside `action`

with_indent: True, succ_cnt: 10/10

with_indent: False, succ_cnt: 10/10

### with `Follow format example's json format`  inside `action`

with_indent: True, succ_cnt: 10/10

with_indent: False, succ_cnt: 10/10

```

### CODE REVIEW

File: metagpt/actions/action_node.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ class ReviseMode(Enum):
 {constraint}
 
 ## action
-generate output and make sure it follows the format example.
+Follow format example's json format, generate output and make sure it follows the format example.
```
Comment (geekan) [inline metagpt/actions/action_node.py:89]: the word "json" or "markdown" depends on user's choice.

File: metagpt/actions/action_node.py [hunk 2]
```diff
@@ -469,7 +469,7 @@ async def auto_review(self, template: str = REVIEW_TEMPLATE) -> dict[str, str]:
             return dict()
 
         prompt = template.format(
-            nodes_output=json.dumps(nodes_output, ensure_ascii=False, indent=4), tag=TAG, constraint=FORMAT_CONSTRAINT
+            nodes_output=json.dumps(nodes_output, ensure_ascii=False), tag=TAG, constraint=FORMAT_CONSTRAINT
```
Comment (geekan) [inline metagpt/actions/action_node.py:472]: could you provide more evidence for this?

File: metagpt/actions/action_node.py (hunk 1 above)
Comment (better629) [reply metagpt/actions/action_node.py:89]: add `prompt_schema` for user's choice

File: metagpt/actions/action_node.py (hunk 2 above)
Comment (better629) [reply metagpt/actions/action_node.py:472]: see `other` part in PR desc

**Review by geekan — APPROVED** [review]

**Review by geekan — APPROVED** [review]
LGTM

**Pull request merged**